hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -562,6 +578,26 @@ private List<BalanceChange> constructBalanceChanges(final CryptoTransferWrapper
                 }
             }
 
+            for (var hbarTransfer : transferOp.transferWrapper().hbarTransfers()) {
+                if (hbarTransfer.sender() != null) {
+                    changes.add(
```

Comment:
Not related to your PR, but I'm wondering whether we need to construct a separate 
`final List<BalanceChange> changes = new ArrayList<>()` on each iteration, add the changes in it only to then add them to the `allChanges` list, instead of directly using the `allChanges` list?

Reply:
This seems like a good optimization.  I can do it while I'm here.